### ISSUE TITLE
Investigate using GDAL driver

### ISSUE THREAD

**Issue by rowanwins** [GeoTIFF/jpeg2000#1]
Perhaps it's feasible to write a node binding to the[ GDAL JPEG2000 driver](https://github.com/OSGeo/gdal/tree/master/gdal/frmts/jpeg2000)?

This has been tried before [elsewhere](https://github.com/naturalatlas/node-gdal) so it's doable, and perhaps more sensible than trying to write our own openJPEG wapper/implementation...

This [tutorial](https://medium.com/@tarkus/how-to-call-c-c-code-from-node-js-86a773033892) provides a bit of info as to how it's done

**Comment by DanielJDufour** [comment]
Good idea.  Perhaps our library can detect if GDAL is installed on the system and use it if that's the case.  Performance would definitely be good that way.

For use in the browser, we'd have to use a JS Port of GDAL.  I'm going to check out https://www.npmjs.com/package/gdal-js and see if that will help.

**Comment by DanielJDufour** [comment]
gdal-js won't work because gdal-js README.md says `In order to limit build size, GDAL is currently built with support for GeoTIFFs and PNGs only.`.  I'm going to see if I can fork it and add jp2 support.

**Comment by rowanwins** [comment]
The only thing about `gdal-js` might be that because it contains far more than we want it might be quite heavy size wise... if we can limit ourselves to just supporting the jpeg2000 driver then presumably that would be a fair bit slimmer.
